lib/widget/confirmation_dialog.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:survey_flutter_ic/theme/colors.dart';
+import 'package:survey_flutter_ic/theme/dimens.dart';
+
+class ConfirmationDialog extends ConsumerStatefulWidget {
+  final String title;
+  final String description;
+  final String positiveActionText;
+  final String negativeActionText;
+  final VoidCallback onConfirmed;
+
+  const ConfirmationDialog({
+    super.key,
+    required this.title,
+    required this.description,
+    required this.positiveActionText,
+    required this.negativeActionText,
+    required this.onConfirmed,
+  });
+
+  @override
+  ConsumerState<ConfirmationDialog> createState() => _ConfirmationDialogState();
+}
+
+class _ConfirmationDialogState extends ConsumerState<ConfirmationDialog> {
```

Comment:
Can we use `StatelessWidget` here since we don't use either `ref` or `state` in this widget?

Reply:
@manh-t Makes sense 🚀